scvi/models/modules.py

```diff
@@ -32,23 +77,33 @@ def __init__(self, n_in: int, n_out: int, n_cat_list: Iterable[int] = None,
         else:
             self.n_cat_list = []
 
+        if not use_batch_norm:
+            n_batch_norm = 0
+
         self.fc_layers = nn.Sequential(collections.OrderedDict(
             [('Layer {}'.format(i), nn.Sequential(
                 nn.Linear(n_in + sum(self.n_cat_list), n_out),
                 # Below, 0.01 and 0.001 are the default values for `momentum` and `eps` from
                 # the tensorflow implementation of batch norm; we're using those settings
                 # here too so that the results match our old tensorflow code. The default
                 # setting from pytorch would probably be fine too but we haven't tested that.
-                nn.BatchNorm1d(n_out, momentum=.01, eps=0.001) if use_batch_norm else None,
+                nn.BatchNorm1d(n_out, momentum=.01, eps=0.001) if n_batch_norm == 1 else None,
+                ConditionalInstanceNormalization(
```

Comment:
I thought we were not using this anymore ? Maybe at some point for ICML submission ?

Reply:
I thought I could leave it to check hyperoptimization of vanilla scvi with this `ConditionalInstanceNormalization`. Maybe it could help scvi to correct batch effect. What do you think?

Reply:
ok that works for me